grpc.go

```diff
@@ -338,6 +338,137 @@ func (a *distributedAccount) SignGRPC(ctx context.Context,
 	return sig, nil
 }
 
+// SignMultiGRPC signs data over GRPC.
+func (a *account) SignMultiGRPC(ctx context.Context,
+	root []byte,
+	domain []byte,
+	accounts []e2wtypes.Account,
+) (
+	[]e2types.Signature,
+	error,
+) {
+	ctx, span := otel.Tracer("wealdtech.go-eth2-wallet-dirk").Start(ctx, "SignMultiGRPC", trace.WithAttributes(
+		attribute.String("wallet", a.wallet.Name()),
+		attribute.String("account", a.Name()),
+	))
+	defer span.End()
+
+	if len(root) != 32 {
+		return nil, errors.New("data must be 32 bytes in length")
+	}
+
+	// Ensure these really are all accounts.
+	for i := range accounts {
+		if _, isAccount := accounts[i].(*account); !isAccount {
+			return nil, errors.New("non-account provided in list")
+		}
+	}
+	req := &pb.MultisignRequest{
+		Requests: make([]*pb.SignRequest, len(accounts)),
+	}
+
+	for i := range accounts {
+		assertedAccount, isAccount := accounts[i].(*account)
+		if !isAccount {
+			return nil, errors.New("account not of required type")
+		}
```

Comment:
Is this necessary, given that this has already been checked above?

Reply:
I removed the check above this as I didn't see the need to iterate over the accounts twice.